tfhe/src/core_crypto/gpu/mod.rs

```diff
@@ -152,11 +148,66 @@ impl CudaStream {
         }
     }
 
+    /// Copies data between two CudaVec, selecting a range of `src` as target
+    ///
+    /// # Safety
+    ///
+    /// - [CudaStream::synchronize] __must__ be called after the copy
+    /// as soon as synchronization is required
+    pub unsafe fn copy_src_range_gpu_to_gpu_async<R, T>(
+        &self,
+        range: R,
+        dest: &mut CudaVec<T>,
+        src: &CudaVec<T>,
+    ) where
+        R: std::ops::RangeBounds<usize>,
+        T: Numeric,
+    {
+        let (start, end) = src.range_bounds_to_start_end(range).into_inner();
```

Comment:
can we maybe use the range.count() method ? I think it should return how many elements are expected to be in the range, and use that invariant afterwards, like if we know the range is empty, skip things and if not we know the range is not empty, I let you judge if it's useful

Reply:
Yes sounds good

Reply:
Hmm actually I don't have a count() method for RangeBounds, only start/end/include, so I left this as it was.